src/command.coffee

```diff
@@ -441,14 +442,77 @@ parseOptions = ->
 
 # The compile-time options to pass to the CoffeeScript compiler.
 compileOptions = (filename, base) ->
-  answer = {
-    filename
+  if opts.transpile
+    # The user has requested that the CoffeeScript compiler also transpile
+    # via Babel. We use Babel as an `optionalDependency`; see
+    # https://docs.npmjs.com/files/package.json#optionaldependencies.
+    try
+      require 'babel-core'
+    catch
+      console.error '''
+        To use --transpile, you must have Babel installed and configured.
+        See http://coffeescript.org/#usage
+      '''
+      process.exit 1
+
+    # We’re giving Babel only a string, not a filename or path to a file, so
+    # it doesn’t know where to search to find a `.babelrc` file or a `babel`
+    # key in a `package.json`. So if `opts.transpile` is an object, use that
+    # as Babel’s options; otherwise figure out what the options should be.
+    unless typeof opts.transpile is 'object'
+      # Find the options based on the path to the file being compiled.
+      cantFindOptions = ->
+        console.error '''
+          To use the transpile option, there must be a .babelrc file
+          (or a package.json file with a "babel" key) in the path of the file
+          to be compiled, or in the path of the current working directory.
+          If you are compiling a string via the Node API, the transpile option
+          must be an object with the options to pass to Babel.
+          See http://coffeescript.org/#usage
+        '''
+        process.exit 1
+
+      checkPath = if filename
+        path.dirname filename
+      else if base
+        base
+      else if process?
+        process.cwd()
+      else
+        cantFindOptions()
+
+      loop
+        try
+          opts.transpile = JSON.parse fs.readFileSync path.join(checkPath, '.babelrc'), 'utf-8'
+          break
+        catch
+          try
+            opts.transpile = JSON.parse(fs.readFileSync(path.join(checkPath, 'package.json'), 'utf-8')).babel
```

Comment:
If we find a package.json _without_ the `babel` key, or with `"babel: null"`, then the search will stop and `opts.transpile` will be `undefined`/`null` and `opts.transpile.transpile` a few lines down will crash. I suggest we only stop the search if `packageJson.babel` is truthy.

Reply:
Okay, how’s the latest commit?